runners/samza/src/main/java/org/apache/beam/runners/samza/metrics/SamzaMetricOpFactory.java

```diff
@@ -53,17 +55,31 @@ public enum OpType {
    * @return a {@link Op} for default transform metric computation
    */
   public static @NonNull <T> Op<T, T, Void> createMetricOp(
+      @NonNull String urn,
       @NonNull String pValue,
       @NonNull String transformName,
       @NonNull OpType opType,
       @NonNull SamzaTransformMetricRegistry samzaTransformMetricRegistry) {
     switch (opType) {
       case INPUT:
+        if (isDataShuffleTransform(urn)) {
+          return new SamzaGBKMetricOp<>(
+              pValue, transformName, opType, samzaTransformMetricRegistry);
+        }
         return new SamzaInputMetricOp(pValue, transformName, samzaTransformMetricRegistry);
```

Comment:
Similar to GBKMetricOp, can we also merge SamzaInputMetricOp and SamzaOutputMetricOp into one?

Reply:
Yup it did make sense to simplify this, done!